pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -217,6 +215,15 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		logger.Errorf("Error while syncing the pipelinerun status: %v", err.Error())
 		return c.finishReconcileUpdateEmitEvents(ctx, pr, before, err)
 	}
+	defer func() {
+		if pr.Status.StartTime == nil || pr.Spec.Timeout == nil {
+			return
+		}
+		// Compute the time since the task started.
+		elapsed := time.Since(pr.Status.StartTime.Time)
+		// Snooze this resource until the timeout has elapsed.
+		c.snooze(pr, pr.Spec.Timeout.Duration-elapsed)
```

Comment:
getPipelineRunTimeout previously checked that pr.Spec.Timeout wasn't nil, and consulted the configmap for a default timeout, how does this code handle the default timeout?

Reply:
Brought back `getPipelineRunTimeout`, but changed its return type.

Reply:
I don't see `getPipelineRunTimeout` in the latest commit (unless I'm just missing it).

I see the nil timeout check above, but we're not falling back to the ConfigMap default with `config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes`. In theory the defaulting webhook should set this for us, but just in case something slips through without a timeout set we should still enforce the default.

Reply:
Nope, I was just missing it 🤦‍♂️ 

Please squash commits before merging 😄

Reply:
Squashed!